### ISSUE TITLE
Support for MAUI

### ISSUE THREAD

**Issue by baddinosaur-kyle** [Laerdal/Laerdal.Dfu#22]
Any chance someone has updated this library to .NET 6 and supporting .NET MAUI (net6.0-android and net6.0-ios)? Is this on the roadmap at all @framinosona?

We've made an attempt at rewriting some of the binding library to handle this on Android in MAUI (and it works!) but we can't get the iOS version working (crashes out when starting the flashing process DFUInstall). We're not iOS developers so are a little out of our depth on it. Any help appreciated!

**Comment by turbo25mike** [comment]
Are you able to share what you've accomplished so far.  I'm looking into rewriting this Library myself and would love to see what you've learned here and possibly collaborate on a new library.

**Comment by baddinosaur-kyle** [comment]
Hey @turbo25mike. Sounds good -- I can share what we have so far. Basically just taken the code and shifted it to .NET 6.0 project type and then dealt with anything that broke. For some reason we can't say the same for the iOS version which is just throwing some exceptions as soon as we try to start it and we just can't figure it out. Do you have any experience in iOS binding libraries?

**Comment by turbo25mike** [comment]
@baddinosaur-kyle I am more of a .NET developer but I'm willing to work with you on this.  I need this working so it's a high priority for me.  I see you forked this but don't see any changes.   Are you planning on making changes in your repo?

**Comment by baddinosaur-kyle** [comment]
@turbo25mike yup, my changes are on 'develop' -- the main branch was behind so decided to fork from there instead as that's where we got it to work. Our hardware to test against is unfortunately out of action for the next few days whilst we wait on a part so I haven't been able to test it (or the Android version since I created this new branch).

**Comment by baddinosaur-kyle** [comment]
@turbo25mike this is the error we're getting on builds for iOS when we're referencing the nuget package we're building with this library, in our app -- this happens at compile time. We used to actually be able to run the app but then it fell over when it was asked to start the firmware flash. Not too sure what this means and can't find many threads to pull on from the internet. We compiled the framework files (IOSDFU and ZipFoundation) via Carthage on a Mac, and grabbed the arm64 framework files so not sure why it wouldn't like them now... only other things that have changed are Xcode versions, of course. All these changes are pushed to develop.

![image](https://user-images.githubusercontent.com/120102912/215828411-c1e367ce-c9f9-4369-98f6-71845400f5af.png)

**Comment by MouratidisA** [comment]
Any news about support for MAUI ??
Do you know a workaround solution that I can use for Nordic device firmware update on MAUI?

**Comment by baddinosaur-kyle** [comment]
@MouratidisA we have it working on our .NET 7 MAUI app for both iOS and Android. It was a few months ago but if I remember right we just did it from source rather than nuget as that was the only way to get it to work on iOS. We installed the dependencies on our Macbook (sharpie and anything else it needs from brew). Changed the target to .NET 7. Then set the flags in the csproj to download native files and to build with sharpie (they're near the top). We referenced the library from our MAUI project and I don't think there was much else to it...
